modules/packages/LinearAlgebra.chpl

```diff
@@ -1883,8 +1883,32 @@ module Sparse {
     return B;
   }
 
-  /* Transpose CSR matrix */
-  proc _array.T where isCSArr(this) { return transpose(this); }
+  /* Transpose Sparse matrix, supports CSR and COO. */
+  proc _array.T where isSparseArr(this) { return transpose(this); }
+
+  pragma "no doc"
+  proc transpose(D: domain) where isSparseDom(D) && !isCSDom(D) {
+    var indices: [1..0] 2*D.idxType;
+    for (i,j) in D {
+      indices.push_back((j,i));
+    }
+    
+    const parentDT = transpose(D.parentDom);
+    var Dom: sparse subdomain(parentDT);
```

Comment:
Replying to your question, this line struck me as wrong. Because we'd want to preserve the distribution/layout of the domain we are transposing. This line, in contrast, creates a COO domain no matter what the layout of the input domain is.

I think we should first check whether the input domain is distributed, and then check its layout. Then, create a distributed/non-distributed domain with the same layout.

Reply:
How would you suggest to approach this? I can think of adding a bunch of if-else cases, but I'm afraid it may be redundant and also miss some cases.